lib/ex_doc/markdown_processor.ex

```diff
@@ -0,0 +1,90 @@
+defmodule ExDoc.MarkdownProcessor do
```

Comment:
We don't need a new module, let's define those callbacks directly on `ExDoc.Markdown`.

Reply:
Hm... Are you sure? Now that you say it I've noticed that the `to_html` function is duplicated here and in the behaviour, which indeed wasteful. But I think that conceptually the two modules serve different functions, even from the point of view of the documentation:

* `ExDoc.Markdown` links to the default supported markdown processors. It's a module that does "markdown stuff", that is, the plumbing necessary to configure and generate markdown, and is highly coupled to `ExDoc`
* `ExDoc.MarkdownProcessor` is a pure behaviour and documents what is expected from the behaviour callbacks. It's a markdown processor, and contains no references to `ExDoc` itself. I also prefer writing:

`@behaviour ExDoc.MarkdownProcessor` in my code rather than `@behaviour ExDoc.Markdown`. In the former it's clear that I am implementing a *markdown processor*, which the `Markdown` module will use.